pkg/controller/deployment_util_test.go

```diff
@@ -80,6 +81,191 @@ var _ = Describe("deployment_util", func() {
+						MaxUnavailable: ptr.To(intstr.FromInt(10)),
+					},
+				},
+			}, 5, 5),
+
+		// Case: InPlaceUpdate strategy with ManualUpdate=true
+		Entry("strategy is InPlaceUpdate with ManualUpdate=true, maxUnavailable=2, replicas=5",
+			machinev1.MachineDeploymentStrategy{
+				Type: machinev1.InPlaceUpdateMachineDeploymentStrategyType,
+				InPlaceUpdate: &machinev1.InPlaceUpdateMachineDeployment{
+					UpdateConfiguration: machinev1.UpdateConfiguration{
+						MaxSurge:       ptr.To(intstr.FromInt(1)),
+						MaxUnavailable: ptr.To(intstr.FromInt(2)),
+					},
+					OrchestrationType: machinev1.OrchestrationTypeManual,
+				},
+			}, 5, 2),
+
+		// Case: InPlaceUpdate strategy with ManualUpdate=false and maxUnavailable > replicas
+		Entry("strategy is InPlaceUpdate with ManualUpdate=false, maxUnavailable=10, replicas=3",
+			machinev1.MachineDeploymentStrategy{
+				Type: machinev1.InPlaceUpdateMachineDeploymentStrategyType,
+				InPlaceUpdate: &machinev1.InPlaceUpdateMachineDeployment{
+					UpdateConfiguration: machinev1.UpdateConfiguration{
+						MaxSurge:       ptr.To(intstr.FromInt(1)),
+						MaxUnavailable: ptr.To(intstr.FromInt(10)),
+					},
+					OrchestrationType: machinev1.OrchestrationTypeAuto,
+				},
+			}, 3, 3),
+
+		// Case: InPlaceUpdate strategy with ManualUpdate=true and replicas=0
+		Entry("strategy is InPlaceUpdate with ManualUpdate=true, replicas=0",
+			machinev1.MachineDeploymentStrategy{
+				Type: machinev1.InPlaceUpdateMachineDeploymentStrategyType,
+				InPlaceUpdate: &machinev1.InPlaceUpdateMachineDeployment{
+					OrchestrationType: machinev1.OrchestrationTypeManual,
+				},
+			}, 0, 0),
+	)
+
+	Describe("#NewISNewReplicas", func() {
+		type data struct {
+			replicas      int
+			oldISReplicas int
+			newISReplicas int
+			expected      int
+			shouldError   bool
+		}
+
+		DescribeTable("#table",
+			func(strategy machinev1.MachineDeploymentStrategy, testData data) {
+				// Setup MachineDeployment and MachineSets
+				deployment := &machinev1.MachineDeployment{
+					Spec: machinev1.MachineDeploymentSpec{
+						Replicas: int32(testData.replicas),
+						Strategy: strategy,
+					},
+				}
+
+				newIS := &machinev1.MachineSet{
+					Spec:   machinev1.MachineSetSpec{Replicas: int32(testData.newISReplicas)},
+					Status: machinev1.MachineSetStatus{Replicas: int32(testData.newISReplicas)},
+				}
+
+				allISs := []*machinev1.MachineSet{
+					newIS,
+					{Spec: machinev1.MachineSetSpec{Replicas: int32(testData.oldISReplicas)}, Status: machinev1.MachineSetStatus{Replicas: int32(testData.oldISReplicas)}},
+				}
+
+				result, err := NewISNewReplicas(deployment, allISs, newIS)
+
+				if testData.shouldError {
+					Expect(err).To(HaveOccurred())
+				} else {
+					Expect(err).NotTo(HaveOccurred())
+					Expect(result).To(Equal(int32(testData.expected)))
+				}
+			},
+
+			// Case: RollingUpdate strategy, scaling up within max surge
+			Entry("RollingUpdate, scale up within max surge",
```

Comment:
Do you want to keep these comments after a description is already present?

Reply:
removed it.